src/App.js

```diff
@@ -1,19 +1,25 @@
-import React from 'react';
+import React, { useState } from 'react';
+import UserContext from "./contexts/UserContext";
 import GlobalStyle from "./styles/global";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import Timeline from './routes/private/Timeline';
 import SignUpRoute from './routes/public/SingUpRoute/SignUpRoute';
 import LogInRoute from './routes/public/LoginRoute/LoginRoute';
 
 function App() {
+
+  const [user, setUser] = useState("");
+
   return (
     <Router>
-      <GlobalStyle />
-      <Switch>
-        <Route exact path="/" component={LogInRoute} />
-        <Route exact path="/sign-up" component={SignUpRoute} />
-        <Route exact path='/timeline' component={Timeline} />
-      </Switch>
+      <UserContext.Provider value={{ user, setUser }}>
+        <GlobalStyle />
+        <Switch>
+          <Route exact path="/" component={LogInRoute} />
+          <Route exact path="/sign-up" component={SignUpRoute} />
+          <Route exact path='/timeline' component={Timeline} />
```

Comment:
Algumas observações: 

A rota "/" deve vir por último. 

Podemos colocar ao final uma rota para redirecionar caso seja digitado qualquer coisa após a / que não seja as rotas existentes, 
  <Redirect to="/sign-up" /> ou 
  <Redirect to ="/" > 
  Isso pq se o usuário digitar /sgin por exemplo, será redirecionado para uma rota existente ;)